src/wrapper/strings/java_str.rs

```diff
@@ -15,10 +16,72 @@ pub struct JavaStr<'a: 'b, 'b> {
 }
 
 impl<'a: 'b, 'b> JavaStr<'a, 'b> {
-    /// Build a `JavaStr` from an object and a reference to the environment. You
-    /// probably want to use `JNIEnv::get_string` instead.
+    /// Get a pointer to the character array beneath a [JString]
+    ///
+    /// The string will be `NULL` terminated and encoded as
+    /// [Modified UTF-8](https://en.wikipedia.org/wiki/UTF-8#Modified_UTF-8) /
+    /// [CESU-8](https://en.wikipedia.org/wiki/CESU-8).
+    ///
+    /// The implementation may either create a copy of the character array for
+    /// the given `String` or it may pin it to avoid it being collected by the
+    /// garbage collector.
+    ///
+    /// Returns a tuple with the pointer and the status of whether the implementation
+    /// created a copy of the underlying character array.
+    ///
+    /// # Warning
+    ///
+    /// The caller must release the array when they are done with it via
+    /// [Self::release_string_utf_chars]
+    ///
+    /// # Safety
+    ///
+    /// The caller must guarantee that the Object passed in is an instance of `java.lang.String`,
+    /// passing in anything else will lead to undefined behaviour (The JNI implementation
+    /// is likely to crash or abort the process).
+    unsafe fn get_string_utf_chars(
+        env: &JNIEnv<'_>,
+        obj: JString<'_>,
+    ) -> Result<(*const c_char, bool)> {
+        non_null!(obj, "get_string_utf_chars obj argument");
+        let mut is_copy: jboolean = 0;
+        let ptr: *const c_char = jni_non_null_call!(
+            env.get_raw(),
+            GetStringUTFChars,
+            obj.into_raw(),
+            &mut is_copy as *mut _
+        );
+
+        let is_copy = is_copy == JNI_TRUE;
+        println!("GetStringUTFChars, is_copy = {}", is_copy);
```

Comment:
This println probably should be removed

Reply:
oh, oops, yeah I was debugging whether I was getting a copy or or pinned string and forgot to remove that!